[SPARK-10577] [PySpark] DataFrame hint for broadcast join

snowmoon-zhang opened the pull request:
https://issues.apache.org/jira/browse/SPARK-10577

snowmoon-zhang committed:
[SPARK-10577] [PySpark] DataFrame hint for broadcast join

https://issues.apache.org/jira/browse/SPARK-10577

AmplabJenkins commented:
Can one of the admins verify this patch?

rxin commented on python/pyspark/sql/functions.py:
```diff
@@ -189,6 +190,14 @@ def approxCountDistinct(col, rsd=None):
 
+@since(1.6)
+def broadcast(df):
+    """Marks a DataFrame as small enough for use in broadcast joins."""
```
can you add a test for this?

Jianfeng-chs replied:
Reynold Xin, I have added one test for this.

rxin commented on python/pyspark/sql/functions.py:
```diff
@@ -189,6 +190,14 @@ def approxCountDistinct(col, rsd=None):
+    """Marks a DataFrame as small enough for use in broadcast joins."""
+
+    sc = SparkContext._active_spark_context
+    return DataFrame(sc._jvm.functions.broadcast(df._jdf),sc._jsc)
```
add space after comma

snowmoon-zhang committed:
[SPARK-10577] [PySpark] DataFrame hint for broadcast join #8777

https://issues.apache.org/jira/browse/SPARK-10577

add space after comma

snowmoon-zhang committed:
[SPARK-10577] [PySpark] DataFrame hint for broadcast join

https://issues.apache.org/jira/browse/SPARK-10577

Adding a test for broadcast join hint

snowmoon-zhang committed:
[SPARK-10663] [ML] [MLLIB] [DOCS] Change test.toDF to test in Spark ML Programming Guide

https://issues.apache.org/jira/browse/SPARK-10663

snowmoon-zhang committed:
Update ml-guide.md

snowmoon-zhang committed:
Merge remote-tracking branch 'apache/master'

snowmoon-zhang committed:
Revert "[SPARK-10663] [ML] [MLLIB] [DOCS] Change test.toDF to test in Spark ML Programming Guide"

This reverts commit b8daa763229b65735640fac35438f34e12a50dbe.

snowmoon-zhang committed:
Revert "Revert "[SPARK-10663] [ML] [MLLIB] [DOCS] Change test.toDF to test in Spark ML Programming Guide""

This reverts commit 198f0e652770569746ffbc3537d91a30dfa2eee1.

snowmoon-zhang committed:
Merge branch 'master' of https://github.com/Jianfeng-chs/spark

Jianfeng-chs commented:
Since messed commits, closing it
